diff --git a/node_modules/shacl-engine/lib/validations/type.js b/node_modules/shacl-engine/lib/validations/type.js
index c8dcb3c..a257f87 100644
--- a/node_modules/shacl-engine/lib/validations/type.js
+++ b/node_modules/shacl-engine/lib/validations/type.js
@@ -28,7 +28,10 @@ function validateClass (classes) {
     for (const classSet of classes) {
       const result = [...types].some(type => classSet.has(type))
 
-      context.test(result, ns.sh.ClassConstraintComponent, { value: context.valueOrNode })
+      context.test(result, ns.sh.ClassConstraintComponent, { 
+        args: { class: classes.flatMap(classSet => [...classSet.values()]) },
+        value: context.valueOrNode 
+      })
     }
   }
 }
